qiskit/assembler/assemble_schedules.py

```diff
@@ -13,213 +13,223 @@
-            ))
+    Args:
+        schedule: Schedule to assemble.
+        instruction_converter: A converter instance which can convert PulseInstructions to
+                               PulseQobjInstructions.
+        run_config: Configuration of the runtime environment.
 
-    qobj_config = PulseQobjConfig(**qobj_config)
+    Returns:
+        A list of converted instructions, the user pulse library dictionary (from pulse name to
+        pulse command), and the maximum number of readout memory slots used by this Schedule.
+    """
+    max_memory_slot = 0
+    qobj_instructions = []
+    user_pulselib = {}
 
-    return PulseQobj(qobj_id=qobj_id,
-                     config=qobj_config,
-                     experiments=experiments,
-                     header=qobj_header)
+    acquire_instruction_map = defaultdict(list)
+    for time, instruction in schedule.instructions:
+
+        if isinstance(instruction, ParametricInstruction):
+            pulse_shape = ParametricPulseShapes(type(instruction.command)).name
+            if pulse_shape not in run_config.parametric_pulses:
+                # Convert to SamplePulse if the backend does not support it
+                instruction = PulseInstruction(instruction.command.get_sample_pulse(),
+                                               instruction.channels[0],
+                                               name=instruction.name)
+
+        if isinstance(instruction, PulseInstruction):
+            name = instruction.command.name
+            if name in user_pulselib and instruction.command != user_pulselib[name]:
+                name = "{0}-{1:x}".format(name, hash(instruction.command.samples.tostring()))
+                instruction = PulseInstruction(
+                    command=SamplePulse(name=name, samples=instruction.command.samples),
+                    name=instruction.name,
+                    channel=instruction.channels[0])
+            # add samples to pulse library
+            user_pulselib[name] = instruction.command
+
+        if isinstance(instruction, AcquireInstruction):
+            max_memory_slot = max(max_memory_slot,
+                                  *[slot.index for slot in instruction.mem_slots])
+            # Acquires have a single AcquireChannel per inst, but we have to bundle them
+            # together into the Qobj as one instruction with many channels
+            acquire_instruction_map[(time, instruction.command)].append(instruction)
+            continue
+
+        if isinstance(instruction, DelayInstruction):
+            # delay instructions are ignored as timing is explicit within qobj
+            continue
+
+        qobj_instructions.append(instruction_converter(time, instruction))
+
+    if acquire_instruction_map:
+        try:
+            _validate_meas_map(acquire_instruction_map, run_config.meas_map)
+        except AttributeError:
+            pass  # No need to validate meas map if it wasn't provided
```

Comment:
Is the expected `AttributeError` from a missing `run_config.meas_map`?  If so, maybe better to replace the `try` with an `if hasattr(run_config, 'meas_map')` to avoid masking errors in `_validate_meas_map`.

Reply:
ohh again, same comment, same good point. I should have scanned for this myself after the last time :o